spec/manifolds/cli_spec.rb

```diff
@@ -3,86 +3,102 @@
 require_relative "../../lib/manifolds/cli"
 require "fileutils"
 require "logger"
+require "fakefs/spec_helpers"
 
 RSpec.describe Manifolds::CLI do
+  include FakeFS::SpecHelpers
+
   let(:project_name) { "commerce" }
   let(:sub_project_name) { "Pages" }
   let(:null_logger) { Logger.new(File::NULL) }
-  let(:bq_service) { instance_double("Manifolds::Services::BigQueryService", "commerce") }
+
+  before do
+    FakeFS do
+      # Set up any template files that need to exist
+      FileUtils.mkdir_p("#{File.dirname(__FILE__)}/../../lib/manifolds/templates")
+      File.write("#{File.dirname(__FILE__)}/../../lib/manifolds/templates/config_template.yml", "vectors:\nmetrics:")
+      File.write("#{File.dirname(__FILE__)}/../../lib/manifolds/templates/vector_template.yml", "attributes:")
+    end
+  end
 
   describe "#init" do
     subject(:cli) { described_class.new(logger: null_logger) }
```

Comment:
Putting the action being tested in a `before` blocks makes tests harder to read because you need to jump back and forth in the file to figure out what's going on.

Reply:
@tylr may disagree with this one. In other non-Ruby projects I've always put the execution logic in the test itself rather than any setup clause. I just thought this was rspec convention.